manager/scheduler/scheduler.go

```diff
@@ -243,10 +243,16 @@ func (s *Scheduler) deleteTask(ctx context.Context, t *api.Task) {
 
 func (s *Scheduler) createOrUpdateNode(n *api.Node) {
 	var resources api.Resources
+	nodeInfo := s.nodeHeap.nodeInfo(n.ID)
 	if n.Description != nil && n.Description.Resources != nil {
 		resources = *n.Description.Resources
+		// reconcile resources
+		for _, task := range nodeInfo.Tasks {
+			reservations := taskReservations(task.Spec)
+			resources.MemoryBytes -= reservations.MemoryBytes
```

Comment:
@stevvooe we didn't write to `Description`. Here, resources amount is copied from `n.Description.Resources` to `nodeInfo.AvailableResources` **by value**. And I think `nodeInfo.AvailableResources` is served as a secondary place for tracking of reservations.